hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/NamenodeHeartbeatService.java

```diff
@@ -339,8 +341,11 @@ protected NamenodeStatusReport getNamenodeStatusReport() {
       // should be required at some point for QoS
       updateSafeModeParameters(serviceURI, report);
 
-      // Read the stats from JMX (optional)
-      updateJMXParameters(webAddress, report);
+      // Read the stats from JMX (optional) using the login user credentials
+      SecurityUtil.doAsLoginUser((PrivilegedExceptionAction<Void>) () -> {
```

Comment:
It feels like we should do this at a higher level eg. NamenodeHeartbeatService.periodicInvoke(). That way the other operations, such as the rpc call to each NN, also run as the login user.

Reply:
Only the JMX call is having issues authenticating, but I agree we should expect all requests to be made in the same context. I'll expand the scope of this.